internal/operator/clusterutilities.go

```diff
@@ -382,6 +395,38 @@ func GetExporterAddon(spec crv1.PgclusterSpec) string {
 	return exporterDoc.String()
 }
 
+func GetPMMContainer(cl *crv1.Pgcluster, name string) string {
+	fmt.Println("Working on PMM", cl.Spec.PMM.Enabled)
+	if !cl.Spec.PMM.Enabled {
+		return ""
+	}
+
+	log.Debugf("pmm enabled for cluster %q", cl.Name)
+
+	pmmTemplateFields := PMMTemplateFields{
+		Enabled:               cl.Spec.PMM.Enabled,
+		PMMImage:              cl.Spec.PMM.Image,
+		PMMServerHost:         cl.Spec.PMM.ServerHost,
+		PMMServerUser:         cl.Spec.PMM.ServerUser,
+		PMMSecret:             cl.Spec.PMM.PMMSecret,
+		PMMContainerResources: GetResourcesJSON(cl.Spec.PMM.Resources, cl.Spec.PMM.Limits),
+		RootSecretName:        crv1.UserSecretName(cl, crv1.PGUserSuperuser),
+		Name:                  name,
+	}
+
+	if CRUNCHY_DEBUG {
+		_ = config.PMMTemplate.Execute(os.Stdout, pmmTemplateFields)
+	}
+
+	doc := bytes.Buffer{}
+	if err := config.PMMTemplate.Execute(&doc, pmmTemplateFields); err != nil {
+		log.Error(err)
+		return ""
+	}
+
+	return doc.String()
```

Comment:
why call `String()` there?
Because then, you cast this function result back to `[]byte` to unmarshall it to Object

Reply:
need it for example here https://github.com/percona/percona-postgresql-operator/blob/K8SPG-25/internal/operator/cluster/clusterlogic.go#L350

Reply:
yea, looks like i've missed it